app/scripts/lib/rpc-method-middleware/handlers/wallet-createSession/handler.ts

```diff
@@ -104,6 +105,15 @@ async function walletCreateSessionHandler(
     return end(new JsonRpcError(5302, 'Invalid sessionProperties requested'));
   }
 
+  let filteredSessionProperties = {};
+  if (sessionProperties && Object.keys(sessionProperties).length > 0) {
```

Comment:
length check necessary given the check on line 104?, or even dropping these guards and doing something like
```
  const filteredSessionProperties = Object.fromEntries(
      Object.entries(sessionProperties ?? {}).filter(([key]) =>
        isKnownSessionPropertyValue(key),
      ),
    );

Reply:
done here: https://github.com/MetaMask/metamask-extension/pull/30949/commits/2ea1fcdcee5b15741ceaa19030caf1a8290a1c31